static/app/views/performance/transactionSummary/teamKeyTransactionButton.tsx

```diff
@@ -76,55 +79,40 @@ class TeamKeyTransactionButton extends Component<Props, State> {
   }
 
   async fetchData() {
+    const {api, organization, project, transactionName} = this.props;
     const keyFetchID = Symbol('keyFetchID');
     this.setState({isLoading: true, keyFetchID});
 
+    let keyedTeams: Set<string> = new Set();
+    let counts: Map<string, number> = new Map();
+
+    let error: string | null = null;
+
     try {
-      const [keyTransactions, counts] = await Promise.all([
-        this.fetchKeyTransactionsData(),
-        this.fetchCountData(),
-      ]);
-      this.setState({
-        isLoading: false,
-        keyFetchID: undefined,
-        error: null,
-        keyedTeams: new Set(keyTransactions.map(({team}) => team)),
-        counts: new Map(counts.map(({team, count}) => [team, count])),
-      });
+      const teams = await fetchTeamKeyTransactions(api, organization.slug, 'myteams');
+
+      keyedTeams = new Set(
+        teams
+          .filter(({keyed}) =>
+            keyed.find(
+              ({project_id, transaction}) =>
+                project_id === String(project) && transaction === transactionName
+            )
+          )
+          .map(({team}) => team)
```

Comment:
curious, do you need to `map(({team}) => team)` or is just the filter enough?

Reply:
Just the filter alone will result in a set of `objects` and we can't check for membership of objects in sets without a reference to the object itself. This map will result in a set of strings so we can check for membership.

Reply:
Just the filter alone will result in a set of `objects` and we can't check for membership of objects in sets without a reference to the object itself. This map will result in a set of strings so we can check for membership.